### PR TITLE
Ensure target group and load balancer names cannot exceed their max length

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Ensure target group and load balancer names cannot exceed their max length
* Expose `*_max_length` variables in case AWS changes their max length in the future
* Override `load_balancer_name`, similar to `target_group_name`

## why
* To prevent exceeding the max lengths of the tg and lb so the module will be applied correctly

## references
* Closes https://github.com/cloudposse/terraform-aws-alb/issues/27
* [API_CreateTargetGroup](https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_CreateTargetGroup.html)
    > This name must be unique per region per account, can have a maximum of 32 characters, must contain only alphanumeric characters or hyphens, and must not begin or end with a hyphen.
* [API_CreateLoadBalancer](https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_CreateLoadBalancer.html)
    > This name must be unique per region per account, can have a maximum of 32 characters, must contain only alphanumeric characters or hyphens, must not begin or end with a hyphen, and must not begin with "internal-".

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -83,15 +90,16 @@ resource "aws_lb" "default" {
...
+  version         = "0.24.1"
+  attributes      = concat(module.this.attributes, ["default"])
+  id_length_limit = var.target_group_name_max_length
+  context         = module.this.context
 }
 
 resource "aws_lb_target_group" "default" {
   count                = module.this.enabled ? 1 : 0
-  name                 = var.target_group_name == "" ? module.default_target_group_label.id : var.target_group_name
+  name                 = var.target_group_name == "" ? module.default_target_group_label.id : substr(var.target_group_name, 0, var.target_group_name_max_length)
```
Comment (jamengual) [inline main.tf:102]: why you need to `substr` if then the length is set the label.id will be of the required length?

**Review by aknysh — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (nitrocode) [reply main.tf:102]: The substr is on the target group name which is used to override the label id

Comment (jamengual) [reply main.tf:102]: got it thanks

**Review by jamengual — APPROVED** [review]

**Pull request merged**